File: list_1/Max_end3.py
'''
Given an array of ints length 3, figure out which is larger, the first or last
element in the array, and set all the other elements to be that value. Return
the changed array.
max_end3([1, 2, 3]) → [3, 3, 3]
max_end3([11, 5, 9]) → [11, 11, 11]
max_end3([2, 11, 3]) → [3, 3, 3]

author LZ-FSDev
see https://codingbat.com/prob/p135290
'''

def max_end3(nums):
    if (nums[0] > nums[len(nums)-1]):
        nums[0:3] = [nums[0], nums[0], nums[0]]
    elif (nums[0] < nums[len(nums)-1]):
        nums[0:3] = [nums[len(nums)-1], nums[len(nums)-1], nums[len(nums)-1]]
    else:
        nums[0:3] = [nums[0], nums[0], nums[0]]
    return nums

array1 = [1, 2, 3]
array2 = [11, 5, 9]
array3 = [2, 11, 3]

print("max_end3(" + str(array1) + ") -> " + str(max_end3(array1)))
print("max_end3(" + str(array2) + ") -> " + str(max_end3(array2)))
print("max_end3(" + str(array3) + ") -> " + str(max_end3(array3)))

File: array_1/MaxEnd3.java
import java.util.Arrays;

/**
 * Given an array of ints length 3, figure out which is larger, the first or
 * last element in the array, and set all the other elements to be that value.
 * Return the changed array.
 * 
 * <ul>
 *  <li>maxEnd3([1, 2, 3]) → [3, 3, 3]
 *  <li>maxEnd3([11, 5, 9]) → [11, 11, 11]
 *  <li>maxEnd3([2, 11, 3]) → [3, 3, 3]
 * </ul>
 * 
 * @author LZ-FSDev
 * @see https://codingbat.com/prob/p146256
 * @since 17.0.1
 * @version 0.0.1
 */
public class MaxEnd3 {
    public static void main(String[] args) {
        int[] array1 = {1, 2, 3};
        int[] array2 = {11, 5, 9};
        int[] array3 = {2, 11, 3};
        System.out.println("MaxEnd3(" + Arrays.toString(array1) + ") -> "
                             + Arrays.toString(maxEnd3(array1)));
        System.out.println("MaxEnd3(" + Arrays.toString(array2) + ") -> "
                             + Arrays.toString(maxEnd3(array2)));
        System.out.println("MaxEnd3(" + Arrays.toString(array3) + ") -> "
                             + Arrays.toString(maxEnd3(array3)));
    }

    /**
     * Given an array of ints length 3, figure out which is larger, the first
     * or last element in the array, and set all the other elements to be that
     * value. Return the changed array.
     * 
     * @param nums Array of integers.
     * @return An array with all indices set to the highest value in nums. 
     * @since 0.0.1
     */
    public static int[] maxEnd3(int[] nums) {
        if (nums[0] > nums[2]) {
            nums[1] = nums[0];
            nums[2] = nums[0];
        } else {
            nums[0] = nums[2];
            nums[1] = nums[2];
        }
        return nums;
    }    
}
